### ISSUE TITLE
ERROR occurs below when starting vscode

### ISSUE THREAD

**Issue by Wenjieiao** [julia-vscode/julia-vscode#3145]
ERROR: LoadError: IOError: connect: no such file or directory (ENOENT)
Stacktrace:
 [1] wait_connected(x::Base.PipeEndpoint)
   @ Sockets D:\Julia-1.8.2\share\julia\stdlib\v1.8\Sockets\src\Sockets.jl:529
 [2] connect
   @ D:\Julia-1.8.2\share\julia\stdlib\v1.8\Sockets\src\Sockets.jl:564 [inlined]
 [3] connect
   @ D:\Julia-1.8.2\share\julia\stdlib\v1.8\Sockets\src\PipeServer.jl:103 [inlined]
 [4] serve(args::String; is_dev::Bool, crashreporting_pipename::String)  
   @ VSCodeServer c:\Users\wenji\.vscode\extensions\julialang.language-julia-1.38.2\scripts\packages\VSCodeServer\src\VSCodeServer.jl:110
 [5] top-level scope
   @ c:\Users\wenji\.vscode\extensions\julialang.language-julia-1.38.2\scripts\terminalserver\terminalserver.jl:39
in expression starting at c:\Users\wenji\.vscode\extensions\julialang.language-julia-1.38.2\scripts\terminalserver\terminalserver.jl:14

What should I do? call for helps!

**Comment by TimG1964** [comment]
I see the same issue, reported [here](https://stackoverflow.com/questions/77954350/ioerror-connect-no-such-file-or-directory-enoent-at-vscode-startup).

> I use VSCode on Win10 with Julia 1.10. Every time I start up VSCode from scratch, before I do anything, I get the following error in the REPL. I’ve had it for a few days but I have no idea what I did to trigger it. How can I make it go away?
> 
>  *  History restored 
> 
> ERROR: LoadError: IOError: connect: no such file or directory (ENOENT)
> Stacktrace:
>  [1] wait_connected(x::Base.PipeEndpoint)
>    @ Sockets C:\Users\TGebbels\AppData\Local\Programs\Julia-1.10.0\share\julia\stdlib\v1.10\Sockets\src\Sockets.jl:528
>  [2] connect
>    @ C:\Users\TGebbels\AppData\Local\Programs\Julia-1.10.0\share\julia\stdlib\v1.10\Sockets\src\Sockets.jl:563 [inlined]
>  [3] connect
>    @ C:\Users\TGebbels\AppData\Local\Programs\Julia-1.10.0\share\julia\stdlib\v1.10\Sockets\src\PipeServer.jl:103 [inlined]
>  [4] serve(args::String; is_dev::Bool, crashreporting_pipename::String)
>    @ VSCodeServer c:\Users\TGebbels\.vscode\extensions\julialang.language-julia-1.66.2\scripts\packages\VSCodeServer\src\VSCodeServer.jl:110
>  [5] top-level scope
>    @ c:\Users\TGebbels\.vscode\extensions\julialang.language-julia-1.66.2\scripts\terminalserver\terminalserver.jl:39
> in expression starting at c:\Users\TGebbels\.vscode\extensions\julialang.language-julia-1.66.2\scripts\terminalserver\terminalserver.jl:14
> julia> 
> Thanks!

**Comment by pfitzseb** [comment]
What version of VS Code are you on?

**Comment by TimG1964** [comment]
Version: 1.86.0 (user setup)
Commit: 05047486b6df5eb8d44b2ecd70ea3bdf775fd937
Date: 2024-01-31T10:28:19.990Z
Electron: 27.2.3
ElectronBuildId: 26495564
Chromium: 118.0.5993.159
Node.js: 18.17.1
V8: 11.8.172.18-electron.0
OS: Windows_NT x64 10.0.22621

**Comment by kschertler** [comment]
I just got the same on a Mac after running some code completely unrelated to IO things - but after restarting the Julia kernel after a crash. So I guess that error is somehow related to restoring the history (as also seen by @TimG1964).

```
julia> 
 *  History restored 

ERROR: LoadError: IOError: connect: no such file or directory (ENOENT)
Stacktrace:
 [1] wait_connected(x::Base.PipeEndpoint)
   @ Sockets ~/.julia/juliaup/julia-1.10.2+0.x64.apple.darwin14/share/julia/stdlib/v1.10/Sockets/src/Sockets.jl:528
 [2] connect
   @ ~/.julia/juliaup/julia-1.10.2+0.x64.apple.darwin14/share/julia/stdlib/v1.10/Sockets/src/Sockets.jl:563 [inlined]
 [3] connect
   @ ~/.julia/juliaup/julia-1.10.2+0.x64.apple.darwin14/share/julia/stdlib/v1.10/Sockets/src/PipeServer.jl:103 [inlined]
 [4] serve(args::String; is_dev::Bool, crashreporting_pipename::String)
   @ VSCodeServer ~/.vscode/extensions/julialang.language-julia-1.75.2/scripts/packages/VSCodeServer/src/VSCodeServer.jl:110
 [5] top-level scope
   @ ~/.vscode/extensions/julialang.language-julia-1.75.2/scripts/terminalserver/terminalserver.jl:45
in expression starting at /Users/me/.vscode/extensions/julialang.language-julia-1.75.2/scripts/terminalserver/terminalserver.jl:20
in expression starting at /Users/me/.vscode/extensions/julialang.language-julia-1.75.2/scripts/terminalserver/terminalserver.jl:20
```

VSCode version:

```
Version: 1.87.2
Commit: 863d2581ecda6849923a2118d93a088b0745d9d6
Date: 2024-03-08T15:21:31.043Z
Electron: 27.3.2
ElectronBuildId: 26836302
Chromium: 118.0.5993.159
Node.js: 18.17.1
V8: 11.8.172.18-electron.0
OS: Darwin x64 23.2.0
```

Julia version:

```
julia> versioninfo()
Julia Version 1.10.2
Commit bd47eca2c8a (2024-03-01 10:14 UTC)
Build Info:
  Official https://julialang.org/ release
Platform Info:
  OS: macOS (x86_64-apple-darwin22.4.0)
  CPU: 8 × Intel(R) Core(TM) i7-8559U CPU @ 2.70GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-15.0.7 (ORCJIT, skylake)
Threads: 8 default, 0 interactive, 4 GC (on 8 virtual cores)
```

**Comment by TimG1964** [comment]
> Is this a normal Julia session or one where you connected to an external REPL?
 For me, it's just a normal Julia session.

**Comment by kschertler** [comment]
> Is this a normal Julia session or one where you connected to an external REPL?

Yes, it's a normal Julia session within VSCode. 

I also checked the related Julia settings within VSCode. It says that persistent session is enabled - but the checkbox is disabled (c.f. below). Also inside the user or workspace json settings there is nothing set to be true w.r.t persistency.

![image](https://github.com/julia-vscode/julia-vscode/assets/53324497/da8363cd-f266-4684-817a-a4711e2d0c28)

**Comment by pfitzseb** [comment]
Yeah, that setting should work either way if you have tmux installed. The "History restored" message will only pop up if VS Code's built-in restore mechanism is at work, which *should* be impossible for the Julia terminal. 
Anyways, can you try disabling the `terminal.integrated.enablePersistentSessions` setting.

**Comment by kschertler** [comment]
OK, I will set `terminal.integrated.enablePersistentSessions` to `false`. Lets see... (this error is not easy reproducible). But thanks anyway, Sebastian.
